### DISCUSSION TITLE
Use PHP 8.1 and change entity structure

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#45182] [Q&A]
Is it possible in PHP 8.1 instead of:

```
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private $id;

    public function getId(): ?int
    {
        return $this->id;
    }
```

Used this?

```
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
	public readonly int $Id;
```

**Comment by stof** [comment]
To be asked in the doctrine/orm repo instead. This is not a question about Symfony but about doctrine.

**Comment by ahundiak** [comment]
As previously mentioned this is a Doctrine question but, for what it is worth, I have been using public properties on Doctrine entities for years without a problem.  Getters and setters are mostly a waste of bytes in this particular context.  Given that Doctrine uses reflection to hydrate entities then I don't see a problem with your approach.  A quick test will reveal if there are any readonly specific issues.

**Comment by karaka200** [comment]
@stof @ahundiak  This is a question about PHP that is to be used in Symfony framework. See this link:

https://www.php.net/releases/8.1/en.php#readonly_properties

**Reply by mbabker** [reply]
Doctrine ORM development and feature support are 100% independent of Symfony.  So, the suggestion to ask this on the Doctrine repo is correct because being able to use PHP 8.1 features in an entity class is going to be based on whether the ORM supports it, not Symfony.

And, as noted in the [ORM 2.11 release announcement](https://www.doctrine-project.org/2022/01/11/orm-2.11.html), readonly properties are supported.
